### PR TITLE
New scripts for making GKE clusters.

### PR DESCRIPTION
**Pull request by scrosby**
## Changes proposed in this PR

- Make a single cluster or make more than one cluster.

## Why are we making these changes?
Prepare for making multiple clusters for the multi-cluster kubernetes support in progress

### CODE REVIEW

File: scheduler/bin/help-make-cluster [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+echo "---- Creating new cluster (please wait 5 minutes)"
+time gcloud container clusters create $CLUSTERNAME --zone --disk-size=20gb --machine-type=g1-small --num-nodes=3 --preemptible  \
+     --enable-autoupgrade --no-enable-basic-auth --no-issue-client-certificate --no-enable-ip-alias --metadata disable-legacy-endpoints=true \
+     --labels longevity=temporary
+
+echo "---- Setting up gcloud credentials"
+# Add credentials to the kubeconfig used by cook.
+(export KUBECONFIG=${COOK_KUBECONFIG} ; gcloud container clusters get-credentials $CLUSTERNAME --region $ZONE)
+# Add credentials to default kubeconfig for convenience purposes.
+gcloud container clusters get-credentials $CLUSTERNAME --region $ZONE
```
[the remark this answers is not in the document]
Comment (scrosby) [reply scheduler/bin/help-make-cluster]: No answer. It worked this way. Didn't try the other way.

File: scheduler/bin/help-make-cluster [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+CLUSTERNAME=$3
+COOK_KUBECONFIG=$4
+echo "---- Building kubernetes cluster for project=$PROJECT zone=$ZONE clustername=$CLUSTERNAME"
+
+# This creates a cluster, with some nodes that are needed for kubernetes management.
+# Create 7 nodes, with three tainted for alpha, 3 untainted and 3 tainted with gamma pool.
+# The untainted ones are for GKE's own uses for its system pods. Also convenient to have around if I'm doing non-pool tests.
+# The second line of flags about upgrades and legacy endpoints is to suppress some warnings.
+echo "---- Creating new cluster (please wait 5 minutes)"
+time gcloud container clusters create $CLUSTERNAME --zone --disk-size=20gb --machine-type=g1-small --num-nodes=3 --preemptible  \
```
[the remark this answers is not in the document]
Comment (scrosby) [reply scheduler/bin/help-make-cluster]: Thats a bug. Thanks.

File: scheduler/bin/help-make-cluster (hunk 1 above)
[the remark this answers is not in the document]
Comment (scrosby) [reply scheduler/bin/help-make-cluster]: It works either way. Changing to be consistent.

**Comment by dposada** [comment]
@scrosby Travis is red

File: scheduler/bin/help-delete-temporary-clusters [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+# Usage: ./bin/make-gke-test-cluster <project> <zone> <clustername>
+#   Configure a kubernetes cluster for running pool-based integration tests and running pools in general.
+#    NOTE: This script labels any clusters it creates and will DELETE old clusters it created.
+#   <project> is a gcloud project.
+#   <zone> can be a zone. E.g., us-central1-a
+#   <clustername> is the name of a cluster. E.g., 'test-cluster-1'
```
Comment (dposada) [inline scheduler/bin/help-delete-temporary-clusters]: This docstring doesn't seem to match the script

File: scheduler/bin/help-make-cluster [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+# Usage: ./bin/make-gke-test-cluster <project> <zone> <clustername>
+#   Configure a kubernetes cluster for running pool-based integration tests and running pools in general.
+#    NOTE: This script labels any clusters it creates and will DELETE old clusters it created.
+#   <project> is a gcloud project.
+#   <zone> can be a zone. E.g., us-central1-a
+#   <clustername> is the name of a cluster. E.g., 'test-cluster-1'
```
Comment (dposada) [inline scheduler/bin/help-make-cluster]: This docstring doesn't seem to match the script

File: scheduler/bin/make-gke-test-cluster [hunk 5]
```diff
@@ -13,41 +13,10 @@ PROJECT=$1
 ZONE=$2
 CLUSTERNAME=$3
 
-echo "---- Building kubernetes cluster for project=$PROJECT zone=$ZONE clustername=$CLUSTERNAME"
+bin/help-delete-temporary-clusters $ZONE
+bin/help-make-cluster $PROJECT $ZONE $CLUSTERNAME .cook_kubeconfig_1
```
Comment (dposada) [inline scheduler/bin/make-gke-test-cluster]: The `help-` at the beginning of these seems confusing. Can we remove it?

File: scheduler/bin/make-gke-test-clusters [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+set -e
+
+PROJECT=$1
+ZONE=$2
+CLUSTERNAME=$3
+
+bin/help-delete-temporary-clusters $ZONE
+
+# Make 1 clusters.
```
Comment (dposada) [inline scheduler/bin/make-gke-test-clusters]: ?

File: scheduler/bin/help-delete-temporary-clusters (hunk 3 above)
Comment (scrosby) [reply scheduler/bin/help-delete-temporary-clusters]: fixed.

File: scheduler/bin/help-make-cluster (hunk 4 above)
Comment (scrosby) [reply scheduler/bin/help-make-cluster]: fixed

File: scheduler/bin/help-make-cluster [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+set -e
+
+PROJECT=$1
+ZONE=$2
+CLUSTERNAME=$3
+COOK_KUBECONFIG=$4
+echo "---- Building kubernetes cluster for project=$PROJECT zone=$ZONE clustername=$CLUSTERNAME"
```
[the remark this answers is not in the document]
Comment (scrosby) [reply scheduler/bin/help-make-cluster]: good idea

File: scheduler/bin/make-gke-test-cluster (hunk 5 above)
Comment (scrosby) [reply scheduler/bin/make-gke-test-cluster]: I want some way to indicate that these are not supposed to be used directly. help-* prefix isn't unreasonable to do this.

File: scheduler/bin/make-gke-test-clusters [hunk 8]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+# Usage: ./bin/make-gke-test-cluster <project> <zone> <clustername>
+#   Configure a kubernetes cluster for running pool-based integration tests and running pools in general.
+#    NOTE: This script labels any clusters it creates and will DELETE old clusters it created.
+#   <project> is a gcloud project.
+#   <zone> can be a zone. E.g., us-central1-a
+#   <clustername> is the name of a cluster. E.g., 'test-cluster-1'
```
[the remark this answers is not in the document]
Comment (scrosby) [reply scheduler/bin/make-gke-test-clusters]: Fixed to describe that it makes two clusters.

File: scheduler/bin/make-gke-test-clusters (hunk 6 above)
Comment (scrosby) [reply scheduler/bin/make-gke-test-clusters]: Changed. Good find.

File: scheduler/bin/make-gke-test-cluster (hunk 5 above)
Comment (dposada) [reply scheduler/bin/make-gke-test-cluster]: Ah, got it. The way we've done this historically is to create bash functions in a separate file, e.g. `util.sh`

File: scheduler/bin/make-gke-test-clusters [hunk 9]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Usage: ./bin/make-gke-test-cluster <project> <zone> <clustername>
```
Comment (dposada) [inline scheduler/bin/make-gke-test-clusters]:
```suggestion
# Usage: ./bin/make-gke-test-clusters <project> <zone> <clustername>
```

File: scheduler/bin/help-make-cluster [hunk 10]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# Usage: ./bin/make-gke-test-cluster <project> <zone> <clustername> <kube_config_file>
```
Comment (dposada) [inline scheduler/bin/help-make-cluster]:
```suggestion
# Usage: ./bin/help-make-cluster <project> <zone> <clustername> <kube_config_file>
```

**Pull request merged**